neurodsp/burst.py

```diff
@@ -0,0 +1,162 @@
+"""
+burst.py
+Detect periods of bursting in EEG waveforms
+"""
+
+from neurodsp import amp_by_time, filt
+import numpy as np
+
+def detect(x, f_range, Fs, algorithm, thresh, magnitudetype='amplitude',
+           min_osc_periods=3, filter_fn=None, filter_kwargs=None, **kwargs):
+    """
+    Detect bursts using one of several methods.
+    
+    Parameters
+    ----------
+    x : array-like 1d
+        voltage time series
+    f_range : (low, high), Hz
+        frequency range for narrowband signal of interest
+    Fs : float
+        The sampling rate
+    algorithm : string
+        Name of algorithm to be used.
+        'deviation' : 
+    thresh : (low, high), units depend on other parameters
+        Threshold value(s) for determining burst
+        NOTE: only one value is needed for 'slopefit'
+    magnitudetype : string in ('power', 'amplitude')
+        metric of magnitude used for thresholding
+    min_osc_periods : float
+        minimum length of an oscillatory period in terms of the period length of f_range[0]
+    filter_fn : filter function with required inputs (x, f_range, Fs, rmv_edge)
+        function to use to filter original time series, x
+    filter_kwargs : dict
+        keyword arguments to the filter_fn
+    Keyword Arguments : 
+        baseline : string in ('median', 'mean')
+            (thresh only) metric to normalize magnitude used for thresholding
+        thresh_bandpow_pc
+            (magnorm only)
+        whitten parameters
+    """
+    
+    # Set default filtering parameters
+    if filter_kwargs is None:
+        filter_kwargs = {}
+
+    # Compute amplitude time series
+    x_amplitude = amp_by_time(x, Fs, f_range, filter_fn=filt.filter, filter_kwargs=filter_kwargs)
+
+    # Set magnitude as power or amplitude
+    if magnitudetype == 'power':
+        x_magnitude = x_amplitude**2
+    elif magnitudetype == 'amplitude':
+        x_magnitude = x_amplitude
+    else:
+        raise ValueError("Invalid 'magnitude' parameter")
+
+    if algorithm == 'deviation':
+        baseline = kwargs['baseline']
+        
+        # Calculate normalized magnitude
+        if baseline == 'median':
+            norm_mag = x_magnitude / np.median(x_magnitude)
+        elif baseline == 'mean':
+            norm_mag = x_magnitude / np.mean(x_magnitude)
+        else:
+            raise ValueError("Invalid 'baseline' parameter")
+
+        if len(thresh) == 2:
+            thresh_lo, thresh_hi = thresh[0], thresh[1]
+        else:
+            raise ValueError("Invalid 'baseline' parameter")
```

Comment:
should this refer to the thresh parameter?

Reply:
Yes. Fixed in next commit